### PR TITLE
CW-1189: Improve Automatic Node Switching

### PR DESCRIPTION
**Pull request by Blazebrain**
Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods
- [ ] Manual tests in accessibility mode (TalkBack on Android) passed

### CODE REVIEW

File: lib/core/node_switching_service.dart [hunk 1]
```diff
@@ -58,8 +70,27 @@ class NodeSwitchingService {
...
     try {
-      final isHealthy = await appStore.wallet!.checkNodeHealth();
+      // Add timeout to prevent hanging on slow network connections
+      final isHealthy = await appStore.wallet!.checkNodeHealth()
+          .timeout(
+            Duration(seconds: 15),
+            onTimeout: () {
+              printV('Health check timed out, considering node unhealthy');
+              return false;
+            },
```
Comment (OmarHatem28) [inline lib/core/node_switching_service.dart:92]: you do realize that this will not apply an actual timeout, right?

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]
also you added more places that might have a hidden edge case that will require more investigation, like the issue Ramo had.
Please simplify the checks, fix the timeout comment, and test it

File: lib/core/node_switching_service.dart (hunk 1 above)
Comment (Blazebrain) [reply lib/core/node_switching_service.dart:92]: Yeah, we're not actually looking to do an actual timeout for the underlying operation, we just want to stop "waiting" for it if it extends past the accepted timeframe
